src/main/java/yahav/photomosaic/PhotoMosaicApplication.java

```diff
@@ -38,19 +43,21 @@ public void start(Stage primaryStage) throws Exception {
 //                (int) image.getHeight());
 //        PixelWriter pixelWriter = wImage.getPixelWriter();
 //        imageView.setImage(wImage);
-        ArrayList<Rectangle2D> rectangles = new ArrayList<Rectangle2D>();
-        for(int x = 0; x < width; x++)
-            for(int y = 0; y < height; y++){
-                Rectangle2D rect = new Rectangle2D(x,y,50, 50);
-                rectangles.add(rect);
+//        ArrayList<Rectangle2D> rectangles = new ArrayList<Rectangle2D>();
+        for(int x = 0; x < width; x+= xInc)
+            for(int y = 0; y < height; y+= yInc){
+                Rectangle2D rect = new Rectangle2D(x,y,xInc, yInc);
```

Comment:
This creates a rectangle for the image, but when I try and loop through it, it prints an infinite amount of colors and doesn't seem to be what I'm looking for, so I'm not so sure how to use it.

Reply:
I really don't understand what you're trying to do here.

Reply:
This method splits the image into 50x50 pixel boxes, I was trying to use rectangle to set each box into the rectangle but I don't think it really works.